integration-test/src/test/java/com/datastrato/gravitino/integration/test/spark/SparkIT.java

```diff
@@ -339,6 +339,56 @@ void testAlterTableRenameColumn() {
     checkTableColumns(tableName, renameColumns, getTableInfo(tableName));
   }
 
+  @Test
+  void testUpdateColumnPosition() {
+    String tableName = "test_column_position";
+    dropTableIfExists(tableName);
+
+    List<SparkColumnInfo> simpleTableColumns =
+        Arrays.asList(
+            SparkColumnInfo.of("id", DataTypes.StringType, ""),
+            SparkColumnInfo.of("name", DataTypes.StringType, ""),
+            SparkColumnInfo.of("age", DataTypes.StringType, ""));
+
+    sql(
+        String.format(
+            "CREATE TABLE %s (id STRING COMMENT '', name STRING COMMENT '', age STRING COMMENT '') USING PARQUET",
+            tableName));
+    checkTableColumns(tableName, simpleTableColumns, getTableInfo(tableName));
+
+    sql(String.format("ALTER TABLE %S ADD COLUMNS (col1 STRING COMMENT '')", tableName));
+    List<SparkColumnInfo> updateColumnPositionCol1 = new ArrayList<>(simpleTableColumns);
+    updateColumnPositionCol1.add(SparkColumnInfo.of("col1", DataTypes.StringType, ""));
+    checkTableColumns(tableName, updateColumnPositionCol1, getTableInfo(tableName));
+
+    sql(String.format("ALTER TABLE %S CHANGE COLUMN col1 col1 STRING FIRST", tableName));
+    List<SparkColumnInfo> updateColumnPositionFirst = new ArrayList<>();
+    updateColumnPositionFirst.add(SparkColumnInfo.of("col1", DataTypes.StringType, ""));
+    updateColumnPositionFirst.addAll(simpleTableColumns);
+    checkTableColumns(tableName, updateColumnPositionFirst, getTableInfo(tableName));
+
+    sql(String.format("ALTER TABLE %S ADD COLUMNS (col2 STRING COMMENT '')", tableName));
+    List<SparkColumnInfo> updateColumnPositionCol2 = new ArrayList<>();
+    updateColumnPositionCol2.add(SparkColumnInfo.of("col1", DataTypes.StringType, ""));
+    updateColumnPositionCol2.addAll(simpleTableColumns);
+    updateColumnPositionCol2.add(SparkColumnInfo.of("col2", DataTypes.StringType, ""));
+    checkTableColumns(tableName, updateColumnPositionCol2, getTableInfo(tableName));
+
+    sql(String.format("ALTER TABLE %S CHANGE COLUMN col2 col2 STRING AFTER col1", tableName));
+    List<SparkColumnInfo> updateColumnPositionAfter = new ArrayList<>();
+    updateColumnPositionAfter.add(SparkColumnInfo.of("col1", DataTypes.StringType, ""));
+    updateColumnPositionAfter.add(SparkColumnInfo.of("col2", DataTypes.StringType, ""));
+    updateColumnPositionAfter.addAll(simpleTableColumns);
+    checkTableColumns(tableName, updateColumnPositionAfter, getTableInfo(tableName));
+
+    sql(String.format("ALTER TABLE %S CHANGE COLUMN col2 col2 STRING", tableName));
```

Comment:
does this sql produce table change event?

Reply:
https://github.com/apache/spark/blob/e170252714e3662c7354321f78a3250114ea7e9e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala#L4337-L4340
1. sparksql supports only the first and after keywords in alterColumn statements.
2. If the first and after keywords are not passed, pisition is none.

So I understand that this sql doesn't actually change the table, but it's hard for me to build gravitino's default position case with sparksql. cc @FANNG1

Reply:
I'll take a look latter

Reply:
I double checked, seems spark just support `AFTER` and `FIRST` in AlterTable, suggest to remove related logic from `SparkIT` and add a check to the return value of `transformColumnPosition` in the process of `TableChange.UpdateColumnPosition`.

Reply:
Should i remove the transformation of default position in `transformColumnPosition `?
Because sparksql cannot pass default position, if the firsr and after keywords are not specified in sparksql, the table is not changed, so tranform default position is not necessary.

@FANNG1

Reply:
> Should i remove the transformation of default position in `transformColumnPosition `? Because sparksql cannot pass default position, if the firsr and after keywords are not specified in sparksql, the table is not changed, so tranform default position is not necessary.
> 
> @FANNG1

It does not seem to be able to be removed. The columnPosition in updateColumnPosition cannot be null, but it can be null in addColumn.

Reply:
yes,  we should keep default position in transformColumnPosition, and do extra check when changing column position